cmd/controller/app/controller.go

```diff
@@ -261,6 +261,7 @@ func buildControllerContextFactory(ctx context.Context, opts *options.Controller
 		return nil, fmt.Errorf("error parsing ACMEHTTP01SolverResourceLimitsMemory: %w", err)
 	}
 
+	ACMEHTTP01SolverRunAsNonRoot := opts.ACMEHTTP01SolverRunAsNonRoot
```

Comment:
nitpick (non-blocking): Do we need this variable given we only use it in one place later in the file?

nitpick (non-blocking): I'd usually start a local variable with a lowercase letter in Golang:

```suggestion
	acmeHTTP01SolverRunAsNonRoot := opts.ACMEHTTP01SolverRunAsNonRoot
```